src/lib/operators/export_binary.cpp

```diff
@@ -130,7 +130,10 @@ void ExportBinary::write_binary(const Table& table, const std::string& filename)
   }
 }
 
-const std::string ExportBinary::name() const { return "ExportBinary"; }
+const std::string& ExportBinary::name() const {
+  static const auto name = std::string{"ExportBinary"};
```

Comment:
Never seen this. The `static` ensures that the string survives the stack cleanup?
I don't like that.

Reply:
It would be nicer to make it `constexpr auto name = std::string{"ExportBinary"}`, but p0980, which allows `constexpr std::string` is [not supported by anyone](https://en.cppreference.com/w/cpp/compiler_support#C.2B.2B2a_library_features) yet.

If we want to return a string reference, we don't have an alternative. The only other option would be to construct an std::string every time this method is called.

Reply:
Actually, we could probably also do `constexpr auto name = "ExportBinary"; return std::string_view{name};` but I am not sure if that is better.

Reply:
I still don't understand how the reference survives to an object that should be immediately freed?!?

Wouldn't it be better to store the string as part of the operator and return sth like `return _operator_name;`.

Reply:
> I still don't understand how the reference survives to an object that should be immediately freed?!?

https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

> Wouldn't it be better to store the string as part of the operator and return sth like return _operator_name;.

You wouldn't want each instance of the operator to have its own instance of the name. Thus, you would make it a static member. Other than breaking the encapsulation of this method, it wouldn't change much.

Reply:
I just believe that you're right here.